### PR TITLE
Updated R scoring server to support standard MLflow models endpoints.

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
Added /invocation and /ping with expected input / output format and behavior.
## How is this patch tested?
R unit test.
 
## Release Notes
R models deployed with MLflow now support the same rest api endpoints as python models.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [x ] Serving
- [ x] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/R/cli.R [hunk 1]
```diff
@@ -24,8 +24,13 @@ mlflow_cli <- function(...,
 
   python <- dirname(python_bin())
   mlflow_bin <- python_mlflow_bin()
+  mlflow_python_bin <- Sys.getenv("MLFLOW_PYTHON_BIN", NA)
```
Comment (aarondav) [inline mlflow/R/mlflow/R/cli.R]: Why was this change needed?

File: mlflow/R/mlflow/R/model-serve.R [hunk 2]
```diff
@@ -157,6 +156,47 @@ serve_handlers <- function(host, port, ...) {
         ))
       )
     },
+    "^/ping" = function(req, model) {
```
Comment (aarondav) [inline mlflow/R/mlflow/R/model-serve.R]: Could we document the fact that the previous endpoints are deprecated (to be removed in MLflow 2.0)?

Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/model-serve.R]: yes we should do that, I'll update the PR.

File: mlflow/R/mlflow/R/cli.R (hunk 1 above)
Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/cli.R]: This is technically a fix for an earlier oss PR. It is needed to propagate path to python binary to R process launched by mlflow cli. I can put this into a separate PR but since it is a small change and I have discovered this while testing this PR I would leave it here. 

Actually, we should also propagate MLFLOW_BIN. I'll update the PR.

**Review by aarondav — APPROVED** [review]

File: mlflow/R/mlflow/R/model-serve.R [hunk 3]
```diff
@@ -157,6 +161,47 @@ serve_handlers <- function(host, port, ...) {
...
+      data_raw <- rawToChar(req$rook.input$read())
+      headers <- strsplit(req$HEADERS, "\n")
+      content_type <- headers$`content-type` %||% "application/json"
+      df <- switch( content_type,
+        "application/json" = parse_json(data_raw, "split"),
+        "application/json; format=pandas-split" = parse_json(data_raw, "split"),
+        "application/json; format=pandas-records" = parse_json(data_raw, "records"),
+        "application/json-numpy-split" = parse_json(data_raw, "split"),
+        "application/json" = parse_json(data_raw, "split"),
+        "text/csv" = utils::read.csv(text = data_raw),
```
Comment (kevinykuo) [inline mlflow/R/mlflow/R/model-serve.R]:
```suggestion
        "text/csv" = utils::read.csv(text = data_raw, stringsAsFactors = FALSE),
```
just to be safe?

**Review by kevinykuo — COMMENTED** [review]
looks good just a couple comments

File: mlflow/R/mlflow/R/model-serve.R (hunk 3 above)
Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/model-serve.R]: Oh, makes sense, nice catch.

File: mlflow/R/mlflow/R/model-serve.R [hunk 4]
```diff
@@ -2,7 +2,12 @@
 
 #' Serve an RFunc MLflow Model
 #'
-#' Serves an RFunc MLflow model as a local web API.
+#' Serves an RFunc MLflow model as a local REST API server. This interface provides similar
+#' functionality to ``mlflow models serve`` cli command, however, it can only be used to deploy
+#' models that include RFunc flavor. The deployed server supports standard mlflow models interface
+#' with /ping and /invocation endpoints. In addition, R function models also support deprecated
+#' /predict endpoint for generating predictions. The /predict endpoint will be removed in the future
+#' versions of mlflow.
```
[the remark this answers is not in the document]
Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/model-serve.R]: Nice catch, thanks!

**Review by kevinykuo — APPROVED** [review]
LGTM!
